core/src/main/java/com/rudderstack/android/sdk/core/RudderDataResidency.java

```diff
@@ -0,0 +1,130 @@
+     *     <li>If dataPlane URL was not passed while SDK initialisation then as a last resort fallback to the default dataPlane URL.</li>
+     * </ul>
+     * <p>
+     * Default Residency server is US.
+     * <p>
+     * Use the below code snippet to set the Data Residency:
+     * <pre>
+     *     {@code
+     *     rudderClient = RudderClient.getInstance(
+     *             this,
+     *             WRITE_KEY,
+     *             RudderConfig.Builder()
+     *                 .withResidencyServer(ResidencyServer.US)
+     *                 .build()
+     *     )
+     *      }
+     * </pre>
+     */
+    void handleDataPlaneUrl() {
+        if (residencyServer == ResidencyServer.US) {
+            handleDefaultServer();
+        } else {
+            handleOtherServer(residencyServer);
+        }
+    }
+
+    /**
+     * Handle all server other than US.
+     * If passed server is not present then fallback to US server.
+     *
+     * @param residencyServer Residency server set while SDK initialisation
+     */
+    @VisibleForTesting
+    void handleOtherServer(@NonNull ResidencyServer residencyServer) {
+        // TODO: Decide upper or lower case
+        String dataResidencyUrl = getDataResidencyUrl(residencyServer.name());
+        if (dataResidencyUrl != null) {
+            config.setDataPlaneUrl(dataResidencyUrl);
+        } else {
+            handleDefaultServer();
+        }
+    }
+
+    /**
+     * Handle default US server
+     *
+     * If default US server is also not present then no need to do any extra things,
+     * as dataPlaneUrl logic is already handled while config object is build.
+     */
+    @VisibleForTesting
+    void handleDefaultServer() {
+        // TODO: Decide upper or lower case
+        String dataResidencyUrl = getDataResidencyUrl(ResidencyServer.US.name());
+        if (dataResidencyUrl != null) {
+            config.setDataPlaneUrl(dataResidencyUrl);
+        }
+    }
+
+    /**
+     * Look for the region in the data residency list.
+     *
+     * @param region Data Residency region set while SDK initialisation.
+     * @return Corresponding residency URL if present else null.
+     */
+    @VisibleForTesting
+    @Nullable
+    String getDataResidencyUrl(@Nullable String region) {
+        if (region != null)
+            if (isNotEmpty(dataResidencyUrls)) {
+                for (String key : dataResidencyUrls.keySet()) {
+                    if (key.equalsIgnoreCase(region)) {
+                        String dataResidencyUrl = dataResidencyUrls.get(key);
+                        if (isNotEmpty(dataResidencyUrl)) {
+                            return processDataPlaneUrl(dataResidencyUrl);
+                        }
+                        break;
+                    }
+                }
+            }
+        return null;
+    }
+
+    @VisibleForTesting
+    @NonNull
+    String processDataPlaneUrl(@NonNull String dataPlaneUrl) {
```

Comment:
What do you think about renaming it to something like `appendSlashToUrl` ?

Reply:
Will surely implement that.